### DISCUSSION TITLE
Question regarding custom logstash pipelines sending to search nodes

### DISCUSSION THREAD

**Discussion by intendedsector** [Security-Onion-Solutions/securityonion#4802] [Unsupported Versions] [answered]
I'm maintaining a distributed environment with 4 search nodes and have been creating custom logstash output configs and adding them to global.sls. I don't fully understand how to specify the host such that they get sent to redis and load balanced appropriately.  I have a simple config that takes anything with a winevt tag and pushes to an so-winevt index. 

The problem is if I specify the hosts then each search node will get each host specified. How do I make sure that each search node inherits just it's IP in the hosts field? I noticed that the default configs (.conf.jinja) seem to do so dynamically but I don't know if just copy pasting those aspects would work. 

Any help would be greatly appreciated either in a simple solution or just generally understanding this aspect of the stack better.

![image](https://user-images.githubusercontent.com/87439522/125660163-bac21a19-bb7c-41e3-9f70-0af44602b6f2.png)

**Comment by weslambert** [comment, marked as answer]
You should look at the default files in `/opt/so/saltstack/default/salt/logstash/pipelines/config/so` and copy/modify one of those.  The jinja contained within those files with the `.jinja` extension is what allows Salt to render the values that you see in the downstream config files.  Make your changes and put the resultant file(s) into `/opt/so/saltstack/local/salt/logstash/pipelines/config/custom`. 

Next, reference your new files so that they are loaded within the appropriate pipeline (ex. manager or search) by copying `/opt/so/saltstack/default/pillar/logstash/manager.sls` (or `search.sls`) to the equivalent `/opt/so/saltstack/local` directory, then restart Logstash.

**Reply by intendedsector** [reply]
Perfect, I migrated all of our custom pipelines yesterday and this did the trick. Thanks!

**Reply by corycandia** [reply]
@intendedsector Would you mind sharing your solution for this?

I think I'm trying to do something very similar.  I have an instance of ElastiFlow running on another VM, and sending data in to the SO manager via REST on 9200.  Their output allows for [specifying](https://docs.elastiflow.com/docs/config_ref#ef_flow_output_elasticsearch_index_template_pipeline_final) template pipeline final, which I'm hoping I can use to direct the input to this custom pipeline and have redis treat it like the other SO data.  I'd rather not leave a ton of non SO data filling up the manager node, I havent gotten a chance to see if curator even trims other (non-SO) data, so I'd like not to fill up my manager before I get this figured out :)

Using redis to spread out the data instead of expensive licensed features is to totally ninja.  God bless SO.
